scout/server/blueprints/cases/views.py

```diff
@@ -1097,7 +1097,7 @@ def mark_causative(institute_id, case_name, variant_id, partial_causative=False)
 
     # send the user back to the case that was marked as solved
     case_url = url_for(".case", institute_id=institute_id, case_name=case_name)
-    return redirect(case_url)
+    return redirect(request.referrer)
```

Comment:
If unmark causative is triggered from the variant page it's better to return that page instead of the case page